src/EasyAbp.BookingService.Domain/EasyAbp/BookingService/AssetOccupancyProviders/AssetOccupancyProviderBase.cs

```diff
@@ -173,22 +172,29 @@ public virtual async Task<bool> CanOccupyByCategoryAsync(OccupyAssetByCategoryIn
                     var assets =
                         await AssetRepository.GetListAsync(x => x.AssetCategoryId == category.Id && !x.Disabled);
 
+                    var periodOccupancyModels = new List<PeriodOccupancyModel>();
                     foreach (var asset in assets)
                     {
                         var periods =
                             await GetCachedAssetDayPeriodsAsync(asset, category, dateGroup.Key, assetDayPeriods);
 
-                        if (!await IsVolumeSufficientAsync(model, periods))
+                        if (await IsVolumeSufficientAsync(model, periods))
                         {
-                            return false;
+                            periodOccupancyModels.AddRange(periods);
```

Comment:
What happens if it returns `false` directly?

Reply:
If there is a asset with insufficient volume and another asset with sufficient volume in same category, the whole bulk occupancy will fail, but it should success.

Reply:
OK. It's my code bug.